packages/viz/src/lib/Layer.ts

```diff
@@ -93,28 +96,20 @@ export class Layer implements StyledLayer {
     const styleField =
       this._style && this._style.field ? [this._style.field] : undefined;
 
-    await this._source.init(styleField);
-
-    const metadata = this._source.getMetadata();
-
-    const styleProps = this._style
-      ? this._style.getLayerProps(this)
-      : undefined;
-
-    // Get properties of the layer
-    const props = this._source.getProps();
+    if (!this._source.isInitialized) {
```

Comment:
When does it happen?

Reply:
I mean, what is the use case to include this condition?

Reply:
this is executed every time we change some layer properties. E.g. when we set a new style or a new `onClick` or `onHover` handler the layer is recreated again and that line is executed

Reply:
Ok!